src/locales/__tests__/getPreciseNumberFormat.test.ts

```diff
@@ -0,0 +1,41 @@
+import { useTranslation } from 'react-i18next'
+import { renderHook } from '@testing-library/react'
+import { getPreciseNumberFormat } from '../getPreciseNumberFormat'
+
+const usePreciseNumberFormat = (props: { value: string }) => {
+  const { t } = useTranslation()
+  const options = getPreciseNumberFormat(props.value)
+  return t('common.valuePair', options)
+}
+
+describe('getPreciseNumberFormat', () => {
+  test('should render precise number with formatting', () => {
+    const { result } = renderHook(usePreciseNumberFormat, {
+      initialProps: { value: '111222333444555666777888999.111222333444555666' },
+    })
+    expect(result.current).toBe('111,222,333,444,555,666,777,888,999.111222333444555666')
```

Comment:
Big problem with this approach: Firefox does not support stringified number in Intl.NumberFormat accurately and returns "111,222,333,444,555,670,000,000,000" (like NodeJS 18)

Reply:
Same issue on Safari

Reply:
I think Firefox fix will come with https://bugzilla.mozilla.org/show_bug.cgi?id=1795756

Reply:
Safari seems to have missed it in https://bugs.webkit.org/show_bug.cgi?id=215438

Reply:
Made it fallback to "111222333444555666777888999.111222333444555666" in Firefox and Safari. E2E tests verify it